inference-engine/thirdparty/clDNN/kernel_selector/core/actual_kernels/reorder/reorder_kernel_bfyx_to_blocked_format.cpp

```diff
@@ -0,0 +1,246 @@
+    jit.AddConstant(MakeJitConstant("TRANS_BUF_SIZE", tile_size * total_lws));
+
+    if (params.output.GetLayout() == DataLayout::fs_b_yx_fsv32) {
+        jit.AddConstant(MakeJitConstant("FS_B_YX_FSV", 1));
+    }
+
+    if (params.output.GetLayout() == DataLayout::bs_fs_yx_bsv16_fsv16 ||
+        params.output.GetLayout() == DataLayout::bs_fs_zyx_bsv16_fsv16) {
+        jit.AddConstant(MakeJitConstant("DOUBLE_BLOCKED_FORMAT", 1));
+        jit.AddConstant(MakeJitConstant("INPUT0_BATCH_SLICE_NUM", CeilDiv(b, fsv_alignment)));
+    }
+
+    // whether F is tile_size-aligned
+    if (f % tile_size == 0) {
+        jit.AddConstant(MakeJitConstant("F_NO_REMAINDER_CONDITION", "(f < INPUT0_FEATURE_NUM)"));
+    } else {
+        jit.AddConstant(MakeJitConstant("F_REMAINDER_SIZE", f % tile_size));
+        jit.AddConstant(MakeJitConstant("F_REMAINDER_CONDITION", "(f >= (INPUT0_FEATURE_NUM - F_REMAINDER_SIZE)) && (f < INPUT0_FEATURE_NUM)"));
+        jit.AddConstant(MakeJitConstant("F_NO_REMAINDER_CONDITION", "(f < (INPUT0_FEATURE_NUM - F_REMAINDER_SIZE))"));
+    }
+
+    // whether x is tile_size-aligned
+    if (x % tile_size != 0) {
+        jit.AddConstant(MakeJitConstant("X_REMAINDER_SIZE", x % tile_size));
+        jit.AddConstant(MakeJitConstant("X_REMAINDER_CONDITION", "(x >= (INPUT0_SIZE_X - X_REMAINDER_SIZE)) && (x < INPUT0_SIZE_X)"));
+        jit.AddConstant(MakeJitConstant("X_NO_REMAINDER_CONDITION", "(x < (INPUT0_SIZE_X - X_REMAINDER_SIZE))"));
+    }
+
+    return jit;
+}
+
+KernelsData ReorderKernel_bfyx_to_blocked_format::GetKernelsData(const Params& params, const optional_params& options) const {
+    assert(params.GetType() == KernelType::REORDER);
+
+    const reorder_params& orgParams = static_cast<const reorder_params&>(params);
+
+    return GetCommonKernelsData(orgParams, options);
+}
+
+bool ReorderKernel_bfyx_to_blocked_format::Validate(const Params& p, const optional_params& o) const {
+    if (!ReorderKernelBase::Validate(p, o)) {
+        return false;
+    }
+
+    const reorder_params& params = static_cast<const reorder_params&>(p);
+    const auto& input = params.inputs[0];
+    const auto& output = params.output;
+
+    if (input.GetDims().size() != output.GetDims().size()) {
+        return false;
+    }
+
+    // padding is not supported
+    if (output.X().pad.before != 0 || output.X().pad.after != 0 ||
+        output.Y().pad.before != 0 || output.Y().pad.after != 0 ||
+        output.Z().pad.before != 0 || output.Z().pad.after != 0 ||
+        output.W().pad.before != 0 || output.W().pad.after != 0 ||
+        output.Feature().pad.before != 0 || output.Feature().pad.after != 0 ||
+        output.Batch().pad.before != 0 || output.Batch().pad.after != 0) {
+        return false;
+    }
+
+    return true;
+}
+
+KernelsPriority ReorderKernel_bfyx_to_blocked_format::GetKernelsPriority(const Params& /*params*/, const optional_params& /*options*/) const {
+    return FORCE_PRIORITY_5;
```

Comment:
why this is 5 while the blocked => nonblocked is 3?

Reply:
I think the fastness is not due to using subgroup read because subgroup read will be using vload in HW operation. I guess that the differnce is coming from whether to use local memory or just registers..